ParquetClassLibrary/Sandbox/RoomRecipe.cs

```diff
@@ -0,0 +1,115 @@
+                          List<EntityID> in_optionallyRequiredPerimeterBlocks = null)
+            : base (All.RoomRecipeIDs, in_id, in_name)
+        {
+            if (in_MinimumWalkableSpaces < All.Recipes.Rooms.MinWalkableSpaces
+                || in_MinimumWalkableSpaces > All.Recipes.Rooms.MaxWalkableSpaces)
+            {
+                throw new ArgumentOutOfRangeException(nameof(in_MinimumWalkableSpaces));
+            }
+            foreach (var floorID in in_optionallyRequiredWalkableFloors ?? Enumerable.Empty<EntityID>())
+            {
+                if (!floorID.IsValidForRange(All.FloorIDs))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(in_optionallyRequiredWalkableFloors));
+                }
+            }
+            foreach (var wallID in in_optionallyRequiredPerimeterBlocks ?? Enumerable.Empty<EntityID>())
+            {
+                if (!wallID.IsValidForRange(All.BlockIDs))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(in_optionallyRequiredPerimeterBlocks));
+                }
+            }
+            if (null == in_requiredFurnishings)
+            {
+                throw new ArgumentNullException(nameof(in_requiredFurnishings));
+            }
+            if (in_requiredFurnishings.Count < 1)
+            {
+                throw new IndexOutOfRangeException(nameof(in_requiredFurnishings));
+            }
+            foreach (var furnishingID in in_requiredFurnishings.Keys)
+            {
+                if (!furnishingID.IsValidForRange(All.FurnishingIDs))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(in_requiredFurnishings));
+                }
+            }
+
+            MinimumWalkableSpaces = in_MinimumWalkableSpaces;
+            RequiredFloors = in_optionallyRequiredWalkableFloors ?? Enumerable.Empty<EntityID>().ToList();
+            RequiredPerimeterBlocks = in_optionallyRequiredPerimeterBlocks ?? Enumerable.Empty<EntityID>().ToList();
+            RequiredFurnishings = in_requiredFurnishings;
+        }
+        #endregion
+
+        /// <summary>
+        /// Determines if the given <see cref="Room"/> conforms to this <see cref="RoomRecipe"/>.
+        /// </summary>
+        /// <param name="in_room">The <see cref="Room"/> to check.</param>
+        /// <returns>
+        /// <c>ture</c> if <paramref name="in_room"/> is an instance of this <see cref="RoomRecipe"/>;
+        /// <c>false</c> otherwise.
+        /// </returns>
+        public bool Matches(Room in_room)
+        {
+            return in_room.WalkableArea.Count >= MinimumWalkableSpaces
+                && RequiredFloors.All(id => in_room.WalkableArea.Select(space => space.Content.Floor.ID).Contains(id))
+                && RequiredPerimeterBlocks.All(id => in_room.Perimeter.Select(space => space.Content.Block.ID).Contains(id))
+                && RequiredFurnishings.All(kvp => in_room.Furnishings.Contains(kvp));
```

Comment:
I think that this means that if a room requires 3 chairs, and you have 4 chairs, it wouldn't contain that KVP and thus fail. Is that okay?

Reply:
Ooooooooo very good catch!  I need to fix that!